zkevm-circuits/src/evm_circuit/execution/signed_comparator.rs

```diff
@@ -0,0 +1,277 @@
+use crate::{
+    evm_circuit::{
+        execution::ExecutionGadget,
+        step::ExecutionState,
+        util::{
+            common_gadget::SameContextGadget,
+            constraint_builder::{
+                ConstraintBuilder, StepStateTransition, Transition,
+            },
+            from_bytes,
+            math_gadget::{ComparisonGadget, IsEqualGadget},
+            select, Word,
+        },
+        witness::{Block, Call, ExecStep, Transaction},
+    },
+    util::Expr,
+};
+
+use bus_mapping::{eth_types::ToLittleEndian, evm::OpcodeId};
+use halo2::{arithmetic::FieldExt, circuit::Region, plonk::Error};
+
+/// Gadget that implements the ExecutionGadget trait to handle the Opcodes SLT
+/// and SGT.
+#[derive(Clone, Debug)]
+pub(crate) struct SignedComparatorGadget<F> {
+    same_context: SameContextGadget<F>,
+
+    a: Word<F>,
+    b: Word<F>,
+
+    sign_check_a: ComparisonGadget<F, 1>,
```

Comment:
```suggestion
    sign_check_a: LtGadget<F, 1>,
```
Since you only need to check if a.cells[0] < 128, you don't need to the extra constraint for equal test